### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by legionus**

### CODE REVIEW

File: pkg/dockerregistry/server/prune/restore.go [hunk 1]
```diff
@@ -217,7 +218,13 @@ func (r *Fsck) Database(namespace string) error {
 				imageDigest, err := digest.ParseDigest(tagEvent.Image)
 				if err == nil {
 					image, err = r.Client.Images().Get(imageDigest.String(), metav1.GetOptions{})
-					if err != nil {
+					switch {
+					case kerrors.IsNotFound(err):
+						image := imageapiv1.Image{}
+						image.Name = imageDigest.String()
+						r.Restore.BrokenImage(image, err)
```
Comment (dmage) [inline pkg/dockerregistry/server/prune/restore.go]: err check?

**Comment by bparees** [comment]
/hold
(for @dmage's question)

**Comment by dmage** [comment]
My question is addressed.
/unhold

**Comment by legionus** [comment]
> you'll still need to remove the WIP if you want this to merge

I know.

**Pull request merged**